cdc/sink/producer/kafka/kafka.go

```diff
@@ -37,11 +37,16 @@ import (
 	"go.uber.org/zap"
 )
 
-const defaultPartitionNum = 4
+const defaultPartitionNum = 3
```

Comment:
Why is this value changed?

Reply:
1. we would use partition-num as the mod for partition dispatch, and `3` is a prime number, much better for hash
2. `3` is a much more reasonable number for replication, and as I know, Kafka would use `KRaft` for consistency protocol, it's good for `raft`
3. `3` would consume less memory usage.